src/nash/mod.rs

```diff
@@ -261,25 +261,37 @@ impl ExchangeAccount for Nash {
         let resp = self.transport.run(req).await;
 
         Ok(
-            Nash::unwrap_response::<nash_protocol::protocol::place_order::LimitOrderResponse>(
+            Nash::unwrap_response::<nash_protocol::protocol::place_order::PlaceOrderResponse>(
                 resp,
             )?
             .into(),
         )
     }
 
-    async fn market_sell(&self, _req: &OpenMarketOrderRequest) -> Result<Order> {
-        let err = MissingImplementationContent {
-            message: String::from("Not supported order type"),
-        };
-        Err(OpenLimitError::MissingImplementation(err))
+    async fn market_sell(&self, req: &OpenMarketOrderRequest) -> Result<Order> {
+        let req: nash_protocol::protocol::place_order::MarketOrderRequest =
+            Nash::convert_market_request(req, nash_protocol::types::BuyOrSell::Sell);
+
+        let resp = self.transport.run(req).await;
+        Ok(
+            Nash::unwrap_response::<nash_protocol::protocol::place_order::PlaceOrderResponse>(
+                resp,
+            )?
+            .into(),
+        )
     }
 
-    async fn market_buy(&self, _req: &OpenMarketOrderRequest) -> Result<Order> {
-        let err = MissingImplementationContent {
-            message: String::from("Not supported order type"),
-        };
-        Err(OpenLimitError::MissingImplementation(err))
+    async fn market_buy(&self, req: &OpenMarketOrderRequest) -> Result<Order> {
+        let req: nash_protocol::protocol::place_order::MarketOrderRequest =
+            Nash::convert_market_request(req, nash_protocol::types::BuyOrSell::Buy);
+
+        let resp = self.transport.run(req).await;
+        Ok(
+            Nash::unwrap_response::<nash_protocol::protocol::place_order::PlaceOrderResponse>(
+                resp,
+            )?
```

Comment:
Can you directly use `val` instead of `OK(val)?`? It looks redundant or I am missing the point.